hail/python/test/hail/matrixtable/test_matrix_table.py

```diff
@@ -2011,13 +2011,6 @@ def assert_contains_node(t, node):
     assert_contains_node(mt, ir.MatrixExplodeRows)
     assert_unique_uids(mt)
 
-    # test MatrixRepartition
-    if not hl.current_backend().requires_lowering:
-        rmt = hl.utils.range_matrix_table(20, 10, 3)
-        mt = rmt.repartition(5)
-        assert_contains_node(mt, ir.MatrixRepartition)
-        assert_unique_uids(mt)
-
```

Comment:
Can we rewrite this and the TableRepartition one to use naive_coalesce, so the handle_randomness implementations still have test coverage?